src/linear_algebra/basic.lean

```diff
@@ -2213,3 +2213,25 @@ rfl
 end general_linear_group
 
 end linear_map
+
+section
+
+variables [ring R] [add_comm_group M] [module R M] (T : M →ₗ[R] M)
+
+/-- any invertible linear map can be written as a linear equivalence -/
+def linear_equiv.of_invertible [invertible T] : M ≃ₗ[R] M :=
+linear_map.general_linear_group.to_linear_equiv (unit_of_invertible T)
+
+lemma linear_equiv.coe_of_invertible [invertible T] :
+  ↑(linear_equiv.of_invertible T) = T := by { ext, refl }
+
+lemma linear_map.to_equiv_symm_eq_inv_of
+  [invertible T] : ⅟T = (linear_equiv.of_invertible T).symm := by { ext, refl }
+
+/-- a linear map `S` commutes with an invertible linear map `T` if and only if
+`(⅟T).comp (S.comp T) = S` -/
+lemma commute_with_invertible_linear_map_iff_conj_eq_self [invertible T] (S : M →ₗ[R] M) :
+  _root_.commute S T ↔ (⅟ T).comp (S.comp T) = S :=
```

Comment:
This should work as well in a general monoid right? If so, I'd say it should go in `algebra/invertible` (note however that this file is synced with `mathlib4`).

Reply:
Yeah, it should work fine with general monoids (I think @alreadydone suggested this as well here https://github.com/leanprover-community/mathlib/pull/18289#discussion_r1088030467). I don't think I can do this at the moment, but I can add a `TODO` to add the generalised result of this for general monoids and add it to `algebra/invertible` to both `mathlib3` and `mathlib4`.

Does that sound good?

Or if you want, I can add the lemma to `mathlib3` and you can add it to `mathlib4`?